Yoast/wordpress-seo
# Support fallbacks for title and meta description

## Context


* Fallbacks for title and meta description where yet unsupported in agnostic analysis. If we create a new post, the SEO title and the meta description should be the default SEO title and the default meta description (as set by the user in the site wide SEO settings). This PR implements that functionality.

## Summary


This PR can be summarized in the following changelog entry:

* Supports fallbacks for title and meta description when they are empty.

## Relevant technical choices:

* For Japanese, we still haven't adjusted the recommended value for meta description length limit when retrieving the excerpt in this PR. This will be addressed in [a separate issue](https://yoast.atlassian.net/browse/LINGO-1267). This decision is taken to avoid making the scope of this issue too big since the locale (an information that is necessary to determine the correct limit) is not yet set in the store.
* Also, the auto-generated excerpt does not take into account the combined length of the date and separator (visible in the Google Preview for post and pages, though not for taxonomies), while the assessment does take this date into account. This will be addressed in [a separate issue](https://yoast.atlassian.net/browse/LINGO-1243).
* Unit tests were not yet added for changes made inside `seo-integration` package in this PR. This is because no testing framework has been set up for this package. [An issue](https://yoast.atlassian.net/browse/LINGO-1270) is created to add one.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

Requirement:
* Install and activate Classic editor

**Test in Posts/Pages**
* Set a default title and meta description for Posts and Pages:
    * Go to Yoast SEO >  Search Appearance > Content Types
    * For both Posts and Pages, set the default title to `%%Title%%`, `%%Separator%%`, `%%Site title%%`
    * For both Posts and Pages, set the default meta description to `%%Excerpt%%`
* Create a post/page
* Add a text at least 300 words.
* Go to Google preview and confirm that:
    * The title shows your post title, the separator, and the site title.
    * The meta description field should show the first 156 characters of your text (it will be truncated to the space before the 156th character), and confirm that the date is also previewed.
    * The SEO title width assessment bullet and progress bar should be green. 
    * The meta description length assessment bullet and progress bar should be green. Dependent upon your text entered for the post/page, the bullet and bar could also be orange, as it takes into account the length of the date and separator. See the relevant technical choices above. 

**Test in a taxonomy page**
* Set a default title and meta description for taxonomy page:
    * Go to Yoast SEO >  Search Appearance > Taxonomies
    * Set the default title to `%%Term title%%`, `%%Separator%%`, `%%Site title%%`
    * Set the default meta description to `%%Excerpt%%`
* Create a category/tag
* Add a text at least 300 words.
* Go to Google preview and confirm that:
    * The title shows your term title, the separator, and the site title.
    * The meta description field should show the first 156 characters of your text (it will be truncated to the space before the 156th character), and also confirm that the date is NOT previewed.
    * The SEO title width assessment bullet and progress bar should be green. 
    * The meta description length assessment bullet and progress bar should be green.

### Test instructions for QA when the code is in the RC



## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* 

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1223
